legend-engine-xts-sql/legend-engine-xt-sql-postgres-server/src/test/java/org/finos/legend/engine/postgres/PostgresServerAsyncTest.java

```diff
@@ -0,0 +1,201 @@
+        resources = pureModelResourceTestRulePair.getTwo();
+    }
+
+    @BeforeClass
+    public static void setUp()
+    {
+        //limit postgres event thread pool to 2 threads
+        System.setProperty("io.netty.eventLoopThreads", "2");
+
+        LegendTdsTestClient client = new LegendTdsTestClient(resources)
+        {
+            @Override
+            public InputStream executeQueryApi(String query)
+            {
+                try
+                {
+                    //sleep for 6 seconds because default Postgres connection time out is 5 second
+                    LOGGER.info("Start sleep");
+                    Thread.sleep(6000);
+                    LOGGER.info("Stop sleep");
+                }
+                catch (InterruptedException e)
+                {
+                    throw new RuntimeException(e);
+                }
+                return super.executeQueryApi(query);
+            }
+        };
+        LegendSessionFactory legendSessionFactory = new LegendSessionFactory(new LegendExecutionService(client));
+
+        ServerConfig serverConfig = new ServerConfig();
+        serverConfig.setPort(0);
+
+        testPostgresServer = new TestPostgresServer(serverConfig, legendSessionFactory, (user, connectionProperties) -> new NoPasswordAuthenticationMethod(new AnonymousIdentityProvider()));
+        testPostgresServer.startUp();
+    }
+
+
+    private void executeAsyncTest(String testName, AsyncTest callableSupplier) throws Exception
+    {
+        AtomicInteger counter = new AtomicInteger(numberOfExecutor);
+
+        List<Callable<Void>> testArrayList = new ArrayList<>();
+        for (int i = 0; i < numberOfExecutor; i++)
+        {
+            String testNameWithCounter = testName + "_" + i;
+            testArrayList.add(new Callable<Void>()
+            {
+                @Override
+                public Void call() throws Exception
+                {
+                    try
+                    {
+                        LOGGER.info(testNameWithCounter + ": Start");
+                        callableSupplier.test();
+                        LOGGER.info(testNameWithCounter + ": Complete");
+                        counter.decrementAndGet();
+                        return (Void) null;
+                    }
+                    catch (Throwable e)
+                    {
+                        LOGGER.error(testNameWithCounter + ": Failed", e);
+                        throw e;
+                    }
+                }
+            });
+        }
+        ExecutorService service = Executors.newFixedThreadPool(numberOfConcurrent);
+        List<Future<Void>> futureList = service.invokeAll(testArrayList, 100, TimeUnit.SECONDS);
+
+        for (int i = 0; i < futureList.size(); i++)
+        {
+            Future<Void> voidFuture = futureList.get(i);
+            voidFuture.get();
+        }
+
+        Assert.assertEquals(0, counter.get());
+    }
+
+
+    private static interface AsyncTest
```

Comment:
Is this interface necessary? Can we use `Callable<Void>` instead?

Reply:
done